databuilder/backends/base.py

```diff
@@ -74,6 +111,21 @@ def __init__(self, query, implementation_notes=None):
         self.query = query
         self.implementation_notes = implementation_notes or {}
 
+    def validate_against_table_schema(self, schema):
+        # This is a very crude form of validation: we just check that the SQL string
+        # contains each of the column names as words. But without actually executing the
+        # SQL we can't know what it returns
```

Comment:
lol

Reply:
I originally thought as this as joke solution, and then I realised it might actually be quite helpful in warning us immediately if we add a new column without updating the SQL. I'd be happy to remove it once we have proper validation against the database if you think it's a little too whimsical.

Reply:
No, I think it's gives some value and it's hard to see how it could be wrong.